### PR TITLE
Use sycl::device_ptr where appropriate

### PR DESCRIPTION
**Pull request by masterleinad**
See https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_usm_address_spaces.asciidoc.

### CODE REVIEW

**Review by nliber — APPROVED** [review]

File: core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp [hunk 1]
```diff
@@ -615,9 +615,9 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
     // m_result_ptr yet.
     if (size <= 1) {
       results_ptr =
-          static_cast<sycl::global_ptr<value_type>>(instance.scratch_space(
+          static_cast<sycl::device_ptr<value_type>>(instance.scratch_space(
               sizeof(value_type) * std::max(value_count, 1u)));
-      sycl::global_ptr<value_type> device_accessible_result_ptr =
+      sycl::device_ptr<value_type> device_accessible_result_ptr =
           m_result_ptr_device_accessible ? m_result_ptr : nullptr;
```
Comment (dalg24) [inline core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp:621]: This one is not obvious to me.  Can you explain why this change is the right thing.

Comment (masterleinad) [reply core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp]: You're right, that should not have been changed. Fixed.

**Review by dalg24 — APPROVED** [review]
